app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -3,8 +3,51 @@ kind: registration
 name: Global Registration
 model: ProvisioningTemplate
 -%>
+<%
+    curl_opts = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+%>
+#!/bin/bash
+
+# Template parameters:
+<%= "#  - User: [#{@user.login}]" if @user %>
+<%= "#  - Organization: [#{@organization.name}]" if @organization %>
+<%= "#  - Location: [#{@location.name}]" if @location %>
+<%= "#  - Hostgroup: [#{@hostgroup.name}]" if @hostgroup %>
 
 if ! [ $(id -u) = 0 ]; then
     echo "Please run as root"
     exit 1
 fi
+
+if [ -f /etc/os-release ] ; then
+  . /etc/os-release
+fi
+
+HOSTNAME=$(hostname -f)
+
+echo "#"
+echo "# Running registration"
+echo "# Hostname: $HOSTNAME"
+echo "# OS: $NAME"
+echo "#"
+
+if [ $ID = rhel ] ; then
+    echo "RHEL registration"
+<% if plugin_present?('katello') %>
+    CONSUMER_RPM=$(mktemp).rpm
+
+    curl --insecure --output $CONSUMER_RPM <%= foreman_server_url %>/pub/katello-ca-consumer-latest.noarch.rpm
+    yum localinstall $CONSUMER_RPM -y
+    yum install subscription-manager -y
+    subscription-manager register --org=<%= @organization.label %> --activationkey=key1
+    rm -f $CONSUMER
+<% end %>
+else
+    curl -X POST <%= foreman_server_url %>/api/hosts \
+     <%= curl_opts.join(' ') %> \
+     -d "{\"host\": {\"name\": \"$HOSTNAME\", \"build\": \"false\", \"managed\": \"false\" \
+     <%= ", \\\"organization_id\\\": \\\"#{@organization.id}\\\"" if @organization %> \
+     <%= ", \\\"location_id\\\": \\\"#{@location.id}\\\"" if @location %> \
+     <%= ", \\\"hostgroup_id\\\": \\\"#{@hostgroup.id}\\\"" if @hostgroup %> \
+     }}"
```

Comment:
Wouldn't it be easier to use a Ruby hash and call `.to_json` on it? Assuming that's allowed in safemode.

Reply:
Not sure how to do it in hash way with checking presence of variable.
Plus `to_json` is not available in safe mode (i'm getting `undefined method '#to_json' for Hash::Jail (Hash)` error), hard to say if it is good idea to enable it or not.

Reply:
How about this as something in between?
```
curl <%= foreman_server_url %>/api/host <<PAYLOAD
{"host": "$HOSTNAME", "build": false, "managed": false
<%= , "organization_id": @organization.id if @organization %> 
<%= , "location_id": @location.id if @location %>
<%= , "hostgroup_id": @hostgroup.id if @hostgroup %>
}
PAYLOAD
```